contracts/staking/SimpleStaking.sol

```diff
@@ -119,23 +122,17 @@ contract SimpleStaking is DSMath, Pausable, AbstractGoodStaking, StakingToken{
         if (tokenActual < tokenWithdraw) {
             tokenWithdraw = tokenActual;
         }
-        //uint256 gdInterest =  InterestDistribution.withdrawStakeAndInterest(interestData, msg.sender, _amount);
-        //Since we use generic ERC20 function we can just use its interface
-        //ERC20 goodDollar = ERC20(address(avatar.nativeToken()));
-        fm.mintReward(address(iToken), msg.sender); // send rewards to user
         _burn(msg.sender, _amount); // burn their staking tokens
         _decreaseProductivity(msg.sender, _amount);
-        //require(goodDollar.transfer(msg.sender, gdInterest), "withdraw interest transfer failed");
+        fm.mintReward(address(iToken), msg.sender); // send rewards to user
```

Comment:
hmm maybe we dont need the FM to callback the staking contract. as it doesnt send any new information.
what do you think can we simply call _rewardsMinted() here (formerly userAccounting)?

Reply:
Yes we can but then we need to check somewhere if the stack contract was blacklisted or not plus if it was initialized